sdk-installer/src/run.py

```diff
@@ -13,55 +13,130 @@
 # SPDX-License-Identifier: Apache-2.0
 
 import argparse
+import json
 import os
 import re
 import shutil
 import subprocess
-from typing import Optional
+from abc import ABC, abstractmethod
+from typing import Dict, List, Optional
 
 from velocitas_lib import (
     get_programming_language,
     get_project_cache_dir,
     get_workspace_dir,
+    require_env,
 )
+from velocitas_lib.variables import ProjectVariables
 
 SUPPORTED_LANGUAGES = ["cpp", "python"]
 
 
-def get_required_sdk_version_cpp() -> str:
-    """Return the required version of the C++ SDK.
+class PackageManager(ABC):
+    @abstractmethod
+    def is_package_installed(
+        self, package_name: str, package_version: Optional[str] = None
+    ) -> bool:
+        ...
 
-    Returns:
-        str: The required version.
-    """
-    sdk_version: str = "0.4.1"
-    with open(
-        os.path.join(get_workspace_dir(), "conanfile.txt"), encoding="utf-8"
-    ) as conanfile:
-        for line in conanfile:
-            if line.startswith("vehicle-app-sdk/"):
-                sdk_version = line.split("/", maxsplit=1)[1].split("@")[0].strip()
+    @abstractmethod
+    def get_required_package_version(self, package_name: str) -> int:
+        ...
 
-    return sdk_version
+    @abstractmethod
+    def install_local_package(self, path: str) -> None:
+        ...
 
 
-def get_required_sdk_version_python() -> str:
-    """Return the required version of the Python SDK.
+class Conan(PackageManager):
+    def __init__(self, verbose_logging: bool):
+        self._verbose_logging = verbose_logging
 
-    Returns:
-        str: The required version.
-    """
-    sdk_version: str = "0.13.0"
-    requirements_path = os.path.join(
-        get_workspace_dir(), "app", "requirements-velocitas.txt"
-    )
-    if os.path.exists(requirements_path):
-        with open(requirements_path, encoding="utf-8") as requirements_file:
-            for line in requirements_file:
-                if line.startswith("velocitas_sdk") or line.startswith("velocitas-sdk"):
-                    sdk_version = line.split("==")[1].strip()
+    def is_package_installed(
+        self, package_name: str, package_version: Optional[str] = None
+    ) -> bool:
+        search_pattern = package_name
+        if package_version is not None:
+            search_pattern = f"{search_pattern}@{package_version}"
```

Comment:
Although technically correct I would prefer the following for better readability
```suggestion
            search_pattern = f"{package_name}@{package_version}"
```